SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -121,6 +114,36 @@ uint32 ComponentFactory::FillSchemaObject(Schema_Object* ComponentObject, UObjec
 		}
 	}
 
+	// Same as in Native, we're always replicating all FastArrays on every tick. Note that users can create NetDeltaSerialized structs, only
+	// FastArrays are See usages of FObjectReplicator::ReplicateCustomDeltaProperties.
+	for (int32 CmdIndex = 0; CmdIndex < Changes.RepLayout.Cmds.Num(); ++CmdIndex)
+	{
+		const FRepLayoutCmd& Cmd = Changes.RepLayout.Cmds[CmdIndex];
+		const FRepParentCmd& Parent = Changes.RepLayout.Parents[Cmd.ParentIndex];
+
+		if (Cmd.Type != ERepLayoutCmdType::DynamicArray)
+		{
+			continue;
+		}
+
+		GDK_PROPERTY(ArrayProperty)* ArrayProperty = GDK_CASTFIELD<GDK_PROPERTY(ArrayProperty)>(Cmd.Property);
+
+		// Check if this is a FastArraySerializer array and if so, call our custom delta serialization
+		if (UScriptStruct* NetDeltaStruct = GetFastArraySerializerProperty(ArrayProperty))
+		{
+			SCOPE_CYCLE_COUNTER(STAT_FactoryProcessFastArrayUpdate);
+
+			FSpatialNetBitWriter ValueDataWriter(PackageMap);
+
+			if (FSpatialNetDeltaSerializeInfo::DeltaSerializeWrite(NetDriver, ValueDataWriter, Object, Parent.ArrayIndex, Parent.Property,
```

Comment:
I'm not sure we can reasonably call this every replication tick. In native, it'll only replicate the delta, whereas in Spatial we replicate the whole array every time :( (along with some keys/counters to simulate native callbacks). See `FFastArraySerializer::FastArrayDeltaSerialize`. We could potentially work around this by early outing in `FFastArraySerializer::SpatialFastArrayDeltaSerialize` if the array's IDCounter hasn't changed from the previous replication. Either that, or we embed into the FastArray's dirty function call callbacks into the push replication model's dirty flags. Or we call out that we have a parity difference to native here.